src/data/nat/prime.lean

```diff
@@ -465,6 +465,10 @@ begin
     { exact factors_one }, }
 end
 
+lemma eq_of_eq_count_factors {a b : ℕ} (ha: 0 < a) (hb: 0 < b)
+  (h: ∀ p : ℕ, list.count p a.factors = list.count p b.factors) : a = b :=
```

Comment:
```suggestion
lemma eq_of_eq_count_factors {a b : ℕ} (ha : 0 < a) (hb : 0 < b)
  (h : ∀ p : ℕ, list.count p a.factors = list.count p b.factors) : a = b :=
```

Reply:
I think it might be worth also adding a lemma that says `(h : a.factors ~ b.factors) : a = b` which presumably you can use to prove this one.

Reply:
Thanks, I've added that lemma and used it in the second proof.